src/index.js

```diff
@@ -1,14 +1,20 @@
 import {cleanup} from './pure'
 
 // if we're running in a test runner that supports afterEach
-// then we'll automatically run cleanup afterEach test
+// or teardown then we'll automatically run cleanup afterEach test
 // this ensures that tests run in isolation from each other
 // if you don't like this then either import the `pure` module
 // or set the RTL_SKIP_AUTO_CLEANUP env variable to 'true'.
-if (typeof afterEach === 'function' && !process.env.RTL_SKIP_AUTO_CLEANUP) {
-  afterEach(async () => {
-    await cleanup()
-  })
+if (!process.env.RTL_SKIP_AUTO_CLEANUP) {
+  if (typeof afterEach === 'function') {
+    afterEach(async () => {
+      await cleanup()
+    })
+  } else if (typeof teardown === 'function') {
+    teardown(async () => {
```

Comment:
I think you can just disable `no-undef` for this line and add an explainer that eslint doesn't support `typeof` checks:
```suggestion
    // Block is guarded by `typeof` check.
    // eslint does not support `typeof` guards.
    // eslint-disable-next-line no-undef
    teardown(async () => {
```